metamorph/processor_response_map.go

```diff
@@ -53,25 +53,26 @@ func NewProcessorResponseMap(expiry time.Duration) *ProcessorResponseMap {
 
 func (m *ProcessorResponseMap) logWriter() {
 	dir := path.Dir(m.logFile)
-	_ = os.MkdirAll(dir, 0777)
-
-	f, err := os.OpenFile(m.logFile, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	err := os.MkdirAll(dir, 0750)
+	if err != nil {
+		log.Fatalf("failed to create folder for logging %s", err)
+	}
+	f, err := os.OpenFile(m.logFile, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0600)
 	if err != nil {
-		log.Printf("error opening log file: %s", err.Error())
+		log.Fatalf("error opening log file: %s", err)
 	}
 	defer f.Close()
 
 	for prl := range m.logWorker {
-
 		var b []byte
 		b, err = json.Marshal(prl)
 		if err != nil {
-			log.Printf("error marshaling log data: %s", err.Error())
+			log.Fatalf("error marshaling log data: %s", err)
```

Comment:
I'd leave it at just logging without exit.

Reply:
same here, you can't proceed if there is error with parsing json. The `b` value would be useless.

Reply:
Ok, but then here we should then maybe do this
```
if err != nil {
  log.Printf("error marshaling log data: %s", err.Error())
  continue
}
```

Reply:
yup, will amend